lib/service/connect.go

```diff
@@ -443,12 +443,28 @@ func (process *TeleportProcess) periodicSyncRotationState() error {
 	})
 	defer periodic.Stop()
 
+	errors := utils.NewTimedCounter(process.Clock, defaults.ConnectionErrorMeasurementPeriod)
+
 	for {
 		err := process.syncRotationStateCycle()
 		if err == nil {
 			return nil
 		}
-		process.log.Warningf("Sync rotation state cycle failed: %v, going to retry after ~%v.", err, defaults.HighResPollingPeriod)
+		process.log.WithError(err).Warningf("Sync rotation state cycle failed: %v", err)
+
+		// If we have had a *lot* of failures very recently, then it's likely that our
+		// route to the auth server is gone. If we're using a tunnel then it's possible
+		// that the proxy has been reconfigured and the tunnel address has moved.
+		if count := errors.Push(); count > defaults.MaxConnectionErrorsBeforeRestart {
```

Comment:
Please add a test exercising this condition - we should make sure that it works as advertised, as well as guard against possible regressions.